### ISSUE TITLE
Feature Request: Programming API

### ISSUE THREAD

**Issue by ptpaterson** [fauna-brecht/fauna-schema-migrate#12]
Request is to call `generate` (or `migrate`) step from code with an array of fql snippets, instead of relying on CLI to search files.

Use case: I have a project that builds up the required schema with lots of JS and configuration.  I've been turning it into a list of `CreateOrUpdate` queries to run, but there's no idea of migration.  I am tinkering with exporting a list of the basic snippets, and rather than making my own queries, exporting those to file from which to use this tool.  

It would be handy to add the _generate migrations_ step into my code, then use the CLI to `apply`.

I have the workaround (export my own files programmatically first), and all things are experimental -- I'm not worried about priority.  I haven't even gotten a _clean_ install working :p so once that is up I'll keep digging and learning the layout.  The task running code is super integrated into the CLI UI, so I'm learning all about Ink and Commander to catch up to what the important bits are.

**Comment by fauna-brecht** [comment]
Agreed, it's in the list of possible extensions. In your case I think it would mean to be able to import the library and expose a number of functions. For your use case I think the following adaptions are necessary: 
- add 'getResourcesFromFQLSnippets': The code expects a LoadedResources types so it would probably mean that I add a getResourcesFromFQLSnippets to which you pass a string of FQL snippets. 
- planMigrations currently takes a path, instead it should take the resources 
- expose getResourcesFromFQLSnippets, planMigrations and generateMigrations.

Once that is done you could chain those three functions and get a list of migrations of the TaggedExpression[] type. Makes sense?

**Comment by fauna-brecht** [comment]
I needed this feature myself for unit testing. 
There is a test npm library here: https://www.npmjs.com/package/@fauna-databrecht/staging-fauna-schema-migrate

Which currently exposes:
```
export { getSnippetsFromStrings, getSnippetsFromCode } from './state/from-code'
export { diffSnippets } from './migrations/diff'
export { generateMigrations } from './migrations/generate-migration'
export { generateMigrationLetObject } from './migrations/generate-query'
export { generateApplyQuery } from './migrations/advance'
export { generateRollbackQuery } from './migrations/rollback'
```

From the unit tests, one can see what can be done with it: https://github.com/fauna-brecht/fauna-schema-migrate/tree/staging/tests/api

Probably, other things that need to be exposed could be: 

- selecting the last version out of multiple sets of migrations
- loading migrations from files
- writing migrations to files
